src/Components/Components/src/CascadingParameterState.cs

```diff
@@ -55,7 +54,7 @@ public static IReadOnlyList<CascadingParameterState> FindCascadingParameters(Com
             {
                 // Although not all parameters might be matched, we know the maximum number
                 resultStates ??= new List<CascadingParameterState>(infos.Length - infoIndex);
-                resultStates.Add(new CascadingParameterState(info, supplier));
+                resultStates.Add(new CascadingParameterState(info, supplier, componentState));
```

Comment:
I'm a bit lost here - why would `componentState` be the `key`? From the naming I was guessing that `key` would refer to anything supplied as `@key` but that can't be the case if we use `componentState` here.

Maybe some other naming would clarify this.

Reply:
Yeah, naming is a bit odd here. I chose that in the beginning because I didn't know what I was going to use, but I think using `ComponentState` just makes sense, as we aren't passing anything else.